src/Duende.Bff.DistributedCache/DistributedCacheUserSessionStore.cs

```diff
@@ -0,0 +1,189 @@
+                await DeleteUserSessionAsync(session, cancellationToken);
+            }
+
+            _logger.LogDebug(
+                "No record found in user session store when trying to " +
+                "delete user session for key {key}",
+                key);
+        }
+
+        private async Task DeleteUserSessionAsync(
+            UserSession session,
+            CancellationToken cancellationToken = default)
+        {
+            Identifier key = session.GetKeyIdentifier(_applicationDiscriminator);
+            Identifier subjectId = session.GetSubjectIdentifier(_applicationDiscriminator);
+            Identifier sessionId = session.GetSessionIdentifier(_applicationDiscriminator);
+
+            await Task.WhenAll(
+                _distributedCache
+                    .RemoveAsync(key, cancellationToken),
+                _distributedCache
+                    .RemoveAsync(subjectId, cancellationToken),
+                _distributedCache
+                    .RemoveAsync(sessionId, cancellationToken));
+        }
+
+        /// <inheritdoc/>
+        public async Task DeleteUserSessionsAsync(
+            UserSessionsFilter filter,
+            CancellationToken cancellationToken = default)
+        {
+            IReadOnlyCollection<UserSession> sessions =
+                await GetUserSessionsAsync(filter, cancellationToken);
+
+            await Task.WhenAll(
+                sessions.Select(session => DeleteUserSessionAsync(session, cancellationToken)));
+        }
+
+        /// <inheritdoc/>
+        public async Task<UserSession?> GetUserSessionAsync(
+            string key,
+            CancellationToken cancellationToken = default)
+        {
+            Identifier identifier =
+                new Identifier(_applicationDiscriminator, key, Key);
+
+            UserSession? session =
+                await _distributedCache
+                    .GetUserSessionAsync(identifier, cancellationToken);
+
+            if (session is null)
+            {
+                _logger.LogDebug(
+                    "No record found in user session" +
+                    " store when trying to get user session for key {key}",
+                    key);
+            }
+
+            return session;
+        }
+
+        /// <inheritdoc/>
+        public async Task<IReadOnlyCollection<UserSession>> GetUserSessionsAsync(
+            UserSessionsFilter filter,
+            CancellationToken cancellationToken = default)
+        {
+            filter.Validate();
+
+            UserSession? sessionBySubjectId = null;
+            UserSession? sessionBySessionId = null;
+            if (!string.IsNullOrWhiteSpace(filter.SubjectId))
+            {
+                Identifier identifier =
+                    new Identifier(_applicationDiscriminator, filter.SubjectId, Subject);
+
+                sessionBySubjectId =
+                    await _distributedCache.GetUserSessionAsync(identifier, cancellationToken);
+            }
```

Comment:
This is a limitation at the moment. 
We can only get the most recent session 

Given you store the following sessions:
``` 
            { 
                SubjectId = "sub1",
                SessionId = "sid1_1",
            }
            { 
                SubjectId = "sub1",
                SessionId = "sid1_2",
            }, 
```
sub1 will point to the newest cache entry
sid_1_1 to the old cache entry
sid_1_2 to the new cache entry